### PR TITLE
Add table of contents

### PR DESCRIPTION
**Pull request by Bisaloo**
I use `CONTRIBUTING.md` *a lot* both for my own rulesets but also as a primer for new contributors. But I always have a hard time finding the exact part I am interested in. I added a table of contents for quick access to the relevant sections.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -1,3 +1,35 @@
...
+      * [Wildcards in Targets](#wildcards-in-targets)
+         * [Left-Wildcards](#left-wildcards)
+         * [Edge-Case: Right-Wildcards](#edge-case-right-wildcards)
+      * [Complicated Regex in Rules](#complicated-regex-in-rules)
+      * [Enumerating Subdomains](#enumerating-subdomains)
+      * [Target Ordering](#target-ordering)
+      * [Rule Ordering](#rule-ordering)
+      * [Non-working hosts](#non-working-hosts)
+      * [Ruleset Names](#ruleset-names)
+         * [Filenames](#filenames)
```
Comment (Bisaloo) [inline CONTRIBUTING.md]: Is the `Filenames` section actually supposed to be nested in the `Ruleset Names` section?

Comment (RReverser) [reply CONTRIBUTING.md]: I think so, yeah.

**Comment by Bisaloo** [comment]
@Hainish

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -1,3 +1,35 @@
+* [Welcome!](#welcome)
```
Comment (Hainish) [inline CONTRIBUTING.md]: I would add a `Table of Contents` header above this line, just to make it clear and consistent.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -1,3 +1,35 @@
...
+      * [Regex Conventions](#regex-conventions)
+      * [Snapping Redirects](#snapping-redirects)
+      * [Example: Ruleset before style guidelines are applied](#example-ruleset-before-style-guidelines-are-applied)
+      * [Example: Ruleset after style guidelines are applied, with test URLs](#example-ruleset-after-style-guidelines-are-applied-with-test-urls)
+   * [Removal of Rules](#removal-of-rules)
+      * [Regular Rules](#regular-rules)
+      * [HSTS Preloaded Rules](#hsts-preloaded-rules)
+* [Contributing Code](#contributing-code)
+* [Contributing Translations](#contributing-translations)
+
```
Comment (Hainish) [inline CONTRIBUTING.md]: I'd add a line below this one which is a horizontal ruler, denoted by:

`* * *`

File: CONTRIBUTING.md (hunk 2 above)
Comment (Bisaloo) [reply CONTRIBUTING.md]: Should the table of content list itself?

Comment (Hainish) [reply CONTRIBUTING.md]: I don't think that's necessary.

**Comment by Hainish** [comment]
Looks great.  I see you're signing commits.  Do you want to squash & sign or are you okay with me squashing, which will lose your signature?

**Comment by Bisaloo** [comment]
Rebased and squashed. In the future, feel free to squash, it doesn't really matter.

**Pull request merged**